locallib.php

```diff
@@ -475,7 +477,29 @@ public static function coursedisplaytext($course, $tpl) {
             'IDNUMBER'  => $course->idnumber,
             'CATEGORY'  => $catname,
         );
+        // If we have limits defined, apply them.
+        static::apply_template_limits($replacements, $tpl);
         $displaytext = str_replace(array_keys($replacements), $replacements, $tpl);
         return strip_tags($displaytext);
     }
+
+    public static function apply_template_limits(&$replacements, &$template) {
+        $limitpattern = "{(\d+)}";
+        foreach ($replacements as $pattern => $replace) {
+            $limit = array();
+            if (preg_match("/$pattern$limitpattern/", $template, $limit)) {
+                $replacements[$pattern] = static::truncate($replace, (int) $limit[1]);
+            }
+        }
+        $template = preg_replace("/$limitpattern/", "", $template);
+    }
+
+    public static function truncate($string, $length) {
+        if ($length > 0 && strlen($string) > $length) {
```

Comment:
Somehow it seems that the Moodle string functions did not get included in this commit. Using the pure PHP function actually does cause a problem. A character like `ø` counts as two characters. And if the truncation split falls between them you are left with an uninterpretable character.